### DISCUSSION TITLE
Type error for `0x${string}` using usePrepareContractWrite and Typescript

### DISCUSSION THREAD

**Discussion by petermazzocco** [wevm/wagmi#2225] [Q&A] [answered]
I am trying to set up a simple mint button using wagmi React hooks in Nextjs and TypeScript, I am trying to figure out why the `args: [address]` is giving me this error `Type 'undefined' is not assignable to type '`0x${string}`'`

Here is the full code of the component

```
  const { address } = useAccount();

// ABI for contract
  const abi = [
    {
      inputs: [{ internalType: "address", name: "to", type: "address" }],
      name: "safeMint",
      outputs: [],
      stateMutability: "payable",
      type: "function",
    },
  ] as const;

  // Prepare function
  const { config: mint } = usePrepareContractWrite({
    address: "0xE72112e617AD180d3445D8b9aB2cBb11Dc8A325D",
    abi: abi,
    functionName: "safeMint",
    overrides: { value: BigNumber.from("0") }, // 0.00 Ether
    args: [address],
  });

  // Function to mint the NFT
  const {
    write: mintNFT,
    isLoading: minting,
    isSuccess: minted,
    error: mintingError,
    data: tx,
  } = useContractWrite(mint);
```

I am fairly new to TypeScript but I can't seem to find a solution to properly assign a type to address without additional errors. 

The mintNFT function works properly and the `args: [address]` is properly being passed as an 0x address and I can mint an NFT so I am confused over this

**Comment by amfredfred** [comment]
first you don't need line (`overrides: { value: BigNumber.from("0") },` // 0.00 Ether) unless fund is required for minting tokens

the 'undefined' is not assignable to type '0x${string} warning 
this maybe that you are not authenticated and therefore the address variable from the `useAccount` hook will be undefined,

does your function look like this?    `function safeMint(address to) external payable () {....}`

**Reply by petermazzocco** [reply]
Yes my function is currently function safeMint(address to) external payable () {...}

I am confused about the not authenticated because like I said the function runs properly, and I can mint with out issues when connected via RainbowKit's SDK. All functions run perfectly fine, it's just throwing a type error for the undefined.

And to answer the first line, I'll check that out to see if I need to add that! It's a free mint as I set no value for minting

**Reply by amfredfred** [reply, marked as answer]
okay since minting is working and you are using typescript 
```

  const { config: mint } = usePrepareContractWrite({
    address: "0xE72112e617AD180d3445D8b9aB2cBb11Dc8A325D", **<<= here is likely where the warning is,**
      solutions: this may not be best practice but try it
    1. `const [cotract, setContract] =useState<string | undefined>("0xE72112e617AD180d3445D8b9aB2cBb11Dc8A325D")`
    2. address: ("0xE72112e617AD180d3445D8b9aB2cBb11Dc8A325D" as any)
    ...
  });

 do the (address as any) in the args too like this `args: [address as any]`

```

**Reply by petermazzocco** [reply]
The `as any` solved it!

I was doing `as string` but as any fixed it!

**Reply by 0xbuidler** [reply]
@petermazzocco You can use the type `Address` from wagmi instead of `any` fyi.
